pkg/controller/controller_instance_test.go

```diff
@@ -2983,14 +3046,16 @@ func TestPollServiceInstanceClusterServiceBrokerError(t *testing.T) {
 	assertNumberOfActions(t, kubeActions, 0)
 
 	actions := fakeCatalogClient.Actions()
-	assertNumberOfActions(t, actions, 0)
+	assertNumberOfActions(t, actions, 1)
+	assertUpdateStatus(t, actions[0], instance)
 
 	events := getRecordedEvents(testController)
 
 	expectedEvent := warningEventBuilder(errorPollingLastOperationReason).msg(
 		"Error polling last operation:",
-	).msg("Status: 403; ErrorMessage: <nil>; Description: <nil>; ResponseError: <nil>")
-	if err := checkEvents(events, expectedEvent.stringArr()); err != nil {
+	).msg("Status: 400; ErrorMessage: <nil>; Description: <nil>; ResponseError: <nil>")
+	// Event is sent twice: one for Ready condition and one for Failed
```

Comment:
The reason we see 2 events is that we send separate events for Ready and Failed condition in https://github.com/kubernetes-incubator/service-catalog/blob/621d4e22f5fd6ad137d350e8122b59e01d0c3845/pkg/controller/controller_instance.go#L2710-L2716

It may be better to send only the Failed one, but this is out of scope of this PR.